sdk/src/Core/Amazon.Runtime/Pipeline/Handlers/MetricsHandler.cs

```diff
@@ -40,7 +41,8 @@ public override void InvokeSync(IExecutionContext executionContext)
             var operationName = AWSSDKUtils.ExtractOperationName(executionContext.RequestContext.RequestName);
             var spanName = $"{executionContext.RequestContext.ServiceMetaData.ServiceId}.{operationName}";
             var span = TracingUtilities.CreateSpan(executionContext.RequestContext, spanName);
-
+            var callDurationMetricsMeasurer = MetricsUtilities.MeasureDuration(executionContext.RequestContext, TelemetryConstants.CallDurationMetricName);
```

Comment:
Is there a case where MeasureDuration can throw an exception outside of the try? If so the span.dispose won't be called in the finally and you already created the span.